src/Auth.php

```diff
@@ -93,17 +93,24 @@ public function isUsingBasicAuth() {
      * Stores the AuthOptions and TokenParams arguments as defaults for subsequent authorisations.
      * @param array|null $tokenParams Requested token parameters
      * @param array|null $authOptions Overridable auth options, if you don't wish to use the default ones
-     * @param boolean|null $force Forces generation of a fresh token
      * @return \Ably\Models\TokenDetails The new token
      */
-    public function authorise( $tokenParams = array(), $authOptions = array(), $force = false ) {
+    public function authorise( $tokenParams = array(), $authOptions = array() ) {
 
         if ( !empty( $tokenParams ) ) {
-            $this->defaultAuthoriseTokenParams = array_merge( $this->defaultAuthoriseTokenParams, $tokenParams );
+            $tokenParamsCopy = $tokenParams;
+            if ( isset( $tokenParamsCopy['timestamp'] ) ) unset( $tokenParamsCopy['timestamp'] );
+
+            $this->defaultAuthoriseTokenParams = array_merge( $this->defaultAuthoriseTokenParams, $tokenParamsCopy );
         }
         if ( !empty( $authOptions ) ) {
-            $this->defaultAuthoriseAuthOptions = array_merge( $this->defaultAuthoriseAuthOptions, $authOptions );
+            $authOptionsCopy = $authOptions;
```

Comment:
In fact, no. PHP is totally backwards and arrays (but not objects!) are copied by value http://stackoverflow.com/a/1533214/388994 Hence the name authOptions**Copy**.